Explorer/Assets/DCL/Backpack/BackpackCharacterPreviewController.cs

```diff
@@ -0,0 +1,140 @@
+
+namespace DCL.CharacterPreview
+{
+    public class BackpackCharacterPreviewController : IDisposable
+    {
+        private readonly BackpackCharacterPreviewView view;
+        private readonly ICharacterPreviewFactory previewFactory;
+        private readonly BackpackEventBus backpackEventBus;
+        private readonly CharacterPreviewInputEventBus inputEventBus;
+        private readonly IComponentPoolsRegistry poolsRegistry;
+
+        private World world;
+        private CharacterPreviewController previewController;
+        private CharacterPreviewModel previewModel;
+
+        public BackpackCharacterPreviewController(BackpackCharacterPreviewView view, ICharacterPreviewFactory previewFactory, BackpackEventBus backpackEventBus, IComponentPoolsRegistry poolsRegistry, CharacterPreviewInputEventBus inputEventBus)
+        {
+            this.view = view;
+            this.previewFactory = previewFactory;
+            this.backpackEventBus = backpackEventBus;
+            this.poolsRegistry = poolsRegistry;
+            this.inputEventBus = inputEventBus;
+
+            this.backpackEventBus.EquipEvent += OnEquipped;
+            this.backpackEventBus.UnEquipEvent += OnUnequipped;
+            this.backpackEventBus.FilterCategoryByEnumEvent += OnChangeCategory;
+            this.backpackEventBus.ForceRenderEvent += OnForceRenderChange;
+
+            view.CharacterPreviewInputDetector.OnScrollEvent += OnScroll;
+            view.CharacterPreviewInputDetector.OnDraggingEvent += OnDrag;
+        }
+
+        public void InjectToWorld(ref ArchSystemsWorldBuilder<World> builder, in Entity playerEntity)
+        {
+            world = builder.World;
+            InitializePreviewModel(playerEntity);
+        }
+
+        private void InitializePreviewModel(Entity playerEntity)
+        {
+            Avatar avatar = world.Get<Profile>(playerEntity).Avatar;
+
+            previewModel.BodyShape = avatar.BodyShape;
+            previewModel.HairColor = avatar.HairColor;
+            previewModel.SkinColor = avatar.SkinColor;
+            previewModel.ForceRender = new HashSet<string>(avatar.ForceRender);
+        }
+
+        public void Initialize()
+        {
+            previewController = previewFactory.Create(world, poolsRegistry, (RenderTexture)view.RawImage.texture, inputEventBus);
+            OnModelUpdated();
+        }
+
+        private void OnScroll(PointerEventData pointerEventData)
+        {
+            inputEventBus.OnScroll(pointerEventData);
+        }
+
+        private void OnDrag(PointerEventData pointerEventData)
+        {
+            inputEventBus.OnDrag(pointerEventData);
+        }
+
+        private void OnChangeCategory(AvatarSlotCategoryEnum categoryEnum)
+        {
+            inputEventBus.OnChangeCategoryFocus(categoryEnum);
+        }
+
+        private void OnForceRenderChange(IReadOnlyCollection<string> forceRender)
+        {
+            previewModel.ForceRender = new HashSet<string>(forceRender);
```

Comment:
Avoid allocating new instances: just clear the existing one and add the elements

Reply:
fixed!